theano/tensor/nnet/abstract_conv.py

```diff
@@ -204,18 +206,13 @@ def conv2d_grad_wrt_inputs(output_grad,
         (output channels, input channels, filter rows, filter columns).
         See the optional parameter ``filter_shape``.
 
-    :type output_grad_shape: None, tuple/list of len 4 of int or
-        Constant variable.
-    :param output_grad_shape: The shape of the output_grad parameter.
-        Optional, possibly used to choose an optimal implementation.
-        You can give ``None`` for any element of the list to specify that this
-        element is not known at compile time.
-
-    :type input_shape: tuple/list of len 2 of int or Constant variable.
+    :type input_shape: tuple/list of len 4, with the first two dimensions
+        being None or int or Constant and the last two dimensions being
+        Tensor or int or Constant.
     :param input_shape: The shape (row and column size) of the
         input (upsampled) parameter.
-        Not Optional, since given the output_grad_shape and the subsample values,
-        multiple input_shape may be plausible.
+        Not Optional, since given the output_grad_shape and the subsample
+        values, multiple input_shape may be plausible.
```

Comment:
If I am not wrong in the case `subsample` == 1 there is no ambiguity.
@lamblin do we want to make `input_shape` optional in that case or we don't care?